urllib3/connectionpool.py

```diff
@@ -192,6 +194,8 @@ def __init__(self, host, port=None, strict=False,
         for _ in xrange(maxsize):
             self.pool.put(None)
 
+        self.transport_security_manager = transport_security_manager or TransportSecurityManager()
```

Comment:
Because ConnectionPool can be instantiated without PoolManager (as the tests do, for example), removing this initialization means ConnectionPool and Connection have to be able to function without a TSM (with TSM set to None). This means adding a bunch of conditionals (`if self. transport_security_manager: ...`) instead of having a mandatory no-op TSM. Is that what you prefer?